R/competing_hazards.R

```diff
@@ -16,22 +16,31 @@ CompetingOutcome <- R6::R6Class(
 
       self$target <- individual::Bitset$new(size)
       self$rates <- NULL
+      self$relative_rates <- NULL
     },
     set_rates = function(target, rates){
       stopifnot(target$size() == length(rates))
 
       self$target$copy_from(target)
       self$rates <- rates
     },
```

Comment:
I think a more readable, generalisable - and perhaps less error prone - method would be to explicity accept extra arguments for your outcome when you set rates.

```suggestion
    set_rates = function(target, rates, ...){
      stopifnot(target$size() == length(rates))

      self$target$copy_from(target)
      self$rates <- rates
      self$args <- list(...)
    },
```